lib/flight_scheduler/cli.rb

```diff
@@ -72,6 +72,12 @@ def self.create_command(name, args_str = '')
     MAGIC_BATCH_SLOP = Slop::Options.new.tap do |slop|
       slop.parser.config[:suppress_errors] = true
       slop.string '-N', '--nodes', 'The minimum number of required nodes'
+      slop.string '-a', '--array', <<~EOF
```

Comment:
FYI: Currently if there are multiple `#SBATCH ---array` magic comments then only the latest will be included; as opposed to concatenating them all. This may or may not be considered a problem.

Switching to `slop.array` would implicitly concatenate them.

Reply:
My intention was that if multiple `--array` options are specified only the last one would be considered.

Regarding magic comments, they should as a general rule be overridden by command line options.  Consider a script with the magic comments `#SBATCH --nodes 1024` and `#SBATCH --array 1-65536`.  These comments are intended for production use of the script.  However, whilst developing and debugging the script, the user wants to be able to submit their job with a much smaller run.  They want to be able to `scheduler batch --nodes 1 --array 1-16 my-script.sh`.  If the `--array` command-line option and the magic comment were merged together they would not have the smaller submission that they were expecting.

Also, whilst the format for `--array` is currently a CSV, it won't stay that way for much longer.  See https://github.com/openflighthpc/flight-scheduler-controller/pull/8#discussion_r493053228 for more on that.

Reply:
:+1: